pennylane/variable.py

```diff
@@ -49,61 +49,64 @@
-:attr:`Variable.kwarg_values` dictionary respectively; these are
-then returned by :meth:`Variable.val`, using its ``idx`` value, and, for
+For each successive QNode execution, the user-provided values for the positional and keyword
+arguments are stored in :attr:`VariableRef.positional_arg_values` and
+:attr:`VariableRef.kwarg_values` respectively; the values are
+then returned by :meth:`VariableRef.val`, using the VariableRef's ``idx`` attribute, and, for
 keyword arguments, its ``name``, to return the correct value to the operation.
 
 .. note::
     The :meth:`Operation.parameters() <pennylane.operation.Operation.parameters>`
-    property automates the process of unpacking the Variable value.
-    The attribute :meth:`Variable.val` should not need to be accessed outside of advanced usage.
+    property automates the process of unpacking the VariableRef value.
+    The attribute :meth:`VariableRef.val` should not need to be accessed outside of advanced usage.
 """
 import copy
 
 
-class Variable:
+
+class VariableRef:
     """A reference to dynamically track and update circuit parameters.
 
     Represents a free quantum circuit parameter (with a non-fixed value),
     or a placeholder for data/other hard-coded data.
 
     Each time the circuit is executed, it is given a vector of flattened positional argument values,
     and a dictionary mapping keyword-only argument names to vectors of their flattened values.
-    Each element of these vectors corresponds to a Variable instance.
-    Positional arguments are represented by nameless Variables, whereas for keyword-only
-    arguments :attr:`Variable.name` contains the argument name.
-    In both cases :attr:`Variable.idx` is an index into the argument value vector.
+    Each element of these vectors corresponds to a VariableRef instance.
+    Positional arguments are represented by nameless VariableRefs, whereas for keyword-only
+    arguments :attr:`VariableRef.name` contains the argument name.
+    In both cases :attr:`VariableRef.idx` is an index into the argument value vector.
 
-    The Variable has an optional scalar multiplier for the argument it represents.
+    The VariableRef has an optional scalar multiplier for the argument it represents.
 
-    .. note:: Variables currently do not implement any arithmetic
+    .. note:: VariableRefs currently do not implement any arithmetic
         operations other than scalar multiplication.
 
     Args:
         idx  (int): index into the value vector, >= 0
         name (None, str): name of the argument
     """
-
     # pylint: disable=too-few-public-methods
-    free_param_values = (
-        None  #: array[float]: current free parameter values, set in :meth:`~.QNode.evaluate`
-    )
-    kwarg_values = None  #: dict[str->array[float]]: the keyword argument values, set in :meth:`~.QNode.evaluate`
+
+    #: array[float]: current positional parameter values, set in :meth:`.BaseQNode._set_variables`
+    positional_arg_values = None
```

Comment:
Same question here about `#:` syntax above the attribute?

Reply:
It works, and here for some reason the attributes actually appear on the automodapi doc page.
For BaseQNode they don't, and neither do the private methods. Annoying.